connect/runtime/src/test/java/org/apache/kafka/connect/integration/RestartLatchTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.connect.integration;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.apache.kafka.common.utils.MockTime;
+import org.apache.kafka.common.utils.Time;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class RestartLatchTest {
+
+    private Time clock;
+    private RestartLatch latch;
+    private List<RestartLatch> dependents;
+    private AtomicBoolean completed = new AtomicBoolean();
+    private ExecutorService waiters = Executors.newSingleThreadExecutor();
+    private Future<Boolean> future;
+
+    @Before
+    public void setup() {
+        clock = new MockTime();
+    }
+
+    @After
+    public void teardown() {
+        if (waiters != null) {
+            try {
+                waiters.shutdownNow();
+            } finally {
+                waiters = null;
```

Comment:
not required in junit 4 and later. `waiters` get GCed

Reply:
This comment was resolved without a follow up or a change. I know setting to `null` is not required here, but if it's a matter of style or you prefer using this anyway I don't want to insist. But if I've missed something it'd be helpful to know.